template/mergeTranslations.js

```diff
@@ -1,58 +1,35 @@
-/* eslint-env node */
-/* eslint-disable no-console, curly */
-const fs = require('fs');
-const path = require('path');
+import { readdir, readFile, writeFile } from 'node:fs/promises';
+import { dirname, resolve } from 'node:path';
+import { fileURLToPath } from 'node:url';
 
-const basePath = './src/features/translations';
-const tempFolder = `${basePath}/temp`;
-const targetFolder = basePath;
-
-const tempDir = path.resolve(__dirname, tempFolder);
-const targetDir = path.resolve(__dirname, targetFolder);
-
-const getLocalesMap = files => {
-    const localesMap = new Map();
-    files.map(file => {
-        const locale = file.split('.')[1];
-        const storedLocale = localesMap.get(locale);
-        const matchingfiles = storedLocale ? [...storedLocale, file] : [file];
-        localesMap.set(locale, matchingfiles);
-    });
-    return localesMap;
-};
-
-const readDir = dir =>
-    new Promise((resolve, reject) => fs.readdir(dir, (err, files) => (err ? reject(err) : resolve(files))));
+const __dirname = dirname(fileURLToPath(import.meta.url));
 
-const readFile = filePath =>
-    new Promise((resolve, reject) => fs.readFile(filePath, 'utf8', (err, data) => (err ? reject(err) : resolve(data))));
+const basePath = './src/features/translations';
+const tempDir = resolve(__dirname, `${basePath}/temp`);
+const targetDir = resolve(__dirname, basePath);
 
-const writeFile = (filePath, content) =>
-    new Promise((resolve, reject) => fs.writeFile(filePath, content, 'utf8', err => (err ? reject(err) : resolve())));
+// list all files in temp folder
+const allTempFiles = await readdir(tempDir);
 
-const getJsonFileContent = async file => JSON.parse(await readFile(`${tempDir}/${file}`));
+// collect files for each locale
+const filesByLocales = allTempFiles.reduce((map, file) => {
+    const locale = file.split('.')[1];
```

Comment:
We could also write
`const [, locale] = file.split('.');`

Reply:
This is exactly why this shorthand was invented :-) The rest of the script is already elegant except for this ;-)

Reply:
Updated :D